src/irmin-pack/inode.ml

```diff
@@ -300,17 +300,10 @@ struct
   module Compress = struct
     open T
 
-    type name = Indirect of int | Direct of step
-    type address = Indirect of int63 | Direct of H.t
-
-    let address_t : address Irmin.Type.t =
-      let open Irmin.Type in
-      variant "Compress.address" (fun i d -> function
-        | Indirect x -> i x | Direct x -> d x)
-      |~ case1 "Indirect" int63_t (fun x -> Indirect x)
-      |~ case1 "Direct" H.t (fun x -> Direct x)
-      |> sealv
-
+    type dict_key = int [@@deriving irmin]
+    type pack_offset = int63 [@@deriving irmin]
+    type name = Indirect of dict_key | Direct of step
+    type address = Offset of pack_offset | Hash of H.t [@@deriving irmin]
```

Comment:
Are 
```suggestion
    type dict_key = int [@@deriving irmin]
    type pack_offset = int63 [@@deriving irmin]
    type name = Indirect of dict_key | Direct of step
    type address = Offset of pack_offset | Hash of H.t
```

Reply:
`address_t` is needed below in `ptr_t`, AFAICS

Reply:
Ah right, nevermind